src/Momento.Sdk/Internal/Retry/DefaultRetryEligibilityStrategy.cs

```diff
@@ -40,7 +40,31 @@ public class DefaultRetryEligibilityStrategy : IRetryEligibilityStrategy
         private readonly HashSet<Type> _retryableRequestTypes = new HashSet<Type>
         {
             typeof(_SetRequest),
-            typeof(_GetRequest)
+            typeof(_GetRequest),
+            typeof(_DeleteRequest),
+            typeof(_DictionarySetRequest),
+            // not idempotent: typeof(_DictionaryIncrementRequest),
+            typeof(_DictionaryGetRequest),
+            typeof(_DictionaryFetchRequest),
+            typeof(_DictionaryDeleteRequest),
+            typeof(_SetUnionRequest),
+            typeof(_SetDifferenceRequest),
+            typeof(_SetFetchRequest),
+            // not idempotent: typeof(_ListPushFrontRequest),
+            // not idempotent: typeof(_ListPushBackRequest),
+            // not idempotent: typeof(_ListPopFrontRequest),
+            // not idempotent: typeof(_ListPopBackRequest),
+            typeof(_ListFetchRequest),
+            /*
+             *  Warning: in the future, this may not be idempotent
+             *  Currently it supports removing all occurrences of a value.
+             *  In the future, we may also add "the first/last N occurrences of a value".
+             *  In the latter case it is not idempotent.
+             */
+            typeof(_ListRemoveRequest),
```

Comment:
Unlike other commands which are either idempotent or not, this one (in the future) will be idempotent in some cases and not in others.

This is a case where a command can be idempotent depending on a field. If we choose to remove all elements equal to a value, then the command is idempotent. But if we choose to remove, say, the first (last) element equal to a value, then it may not be.

Currently we only support removing all elements equal to a value, hence I have left it retryable.